### PR TITLE
Add WhereAsArray with generic argument

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

File: src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs [hunk 1]
```diff
@@ -259,13 +259,23 @@ public static ImmutableArray<byte> ToImmutable(this MemoryStream stream)
...
+
+        /// <summary>
+        /// Creates a new immutable array based on filtered elements by the predicate. The array must not be null.
+        /// </summary>
+        /// <param name="array">The array to process</param>
+        /// <param name="predicate">The delegate that defines the conditions of the element to search for.</param>
+        public static ImmutableArray<T> WhereAsArray<T, TArg>(this ImmutableArray<T> array, Func<T, TArg, bool> predicate, TArg arg)
+            => WhereAsArrayImpl(array, predicateWithoutArg: null, predicate, arg);
+
+        private static ImmutableArray<T> WhereAsArrayImpl<T, TArg>(this ImmutableArray<T> array, Func<T, bool>? predicateWithoutArg, Func<T, TArg, bool>? predicateWithArg, TArg arg)
```
Comment (cston) [inline src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs]: >this  [](start = 67, length = 5)

It looks like it does not need to be an extension method.

File: src/Workspaces/Core/Portable/Shared/Extensions/ISolutionExtensions.cs [hunk 2]
```diff
@@ -66,16 +66,7 @@ public static Solution WithTextDocumentText(this Solution solution, DocumentId d
...
-            {
-                var document = solution.GetDocument(documentId);
-                if (document != null && document.Project.Language == language)
-                {
-                    yield return documentId;
-                }
-            }
-        }
+        public static ImmutableArray<DocumentId> FilterDocumentIdsByLanguage(this Solution solution, ImmutableArray<DocumentId> documentIds, string language)
+            => documentIds.WhereAsArray((documentId, language) => solution.GetDocument(documentId)?.Project.Language == language, language));
```
Comment (cston) [inline src/Workspaces/Core/Portable/Shared/Extensions/ISolutionExtensions.cs]: >solution [](start = 66, length = 8)

`solution` is captured.

Comment (tmat) [reply src/Workspaces/Core/Portable/Shared/Extensions/ISolutionExtensions.cs]: Ah, I didn't mean to include the IDE changes... removing.

**Comment by tmat** [comment]
Thanks!

**Pull request merged**